src/module/actor/creature/sheet.ts

```diff
@@ -296,6 +297,43 @@ abstract class CreatureSheetPF2e<TActor extends CreaturePF2e> extends ActorSheet
         game.tooltip.activate(anchor, { cssClass: "pf2e carry-type-menu", content, locked: true });
     }
 
+    protected override async _onDrop(event: DragEvent): Promise<boolean | void> {
+        const dataString = event.dataTransfer?.getData("text/plain");
+        const dropData = ((): Record<string, unknown> | null => {
+            try {
+                return JSON.parse(dataString ?? "");
+            } catch {
+                return null;
+            }
+        })();
+        if (!dropData) return;
```

Comment:
Isn't this going to break regular onDrop handling? Might be best to omit this if.

Reply:
This is copied code from Character Sheet's _onDrop - so if it breaks here, wouldn't it break there as well? Are there any valid data payloads being dropped on a sheet that wouldn't be able to be parsed as JSON?

Reply:
I'll give it a shot. I agree that's a better, cleaner way to handle the non-PF2E drops.